src/CovidTracker/calculate_stat_summary.py

```diff
@@ -16,3 +18,17 @@ def calculate_stat_summary(df):
     --------
     >>> calculate_stat_summary(covid_df)
     """
+
+    # Check input parameter validity
+    if not isinstance(df, pd.DataFrame):
+        raise TypeError("Invalid argument type: df must be a pandas DataFrame")
+    elif len(df) == 0:
+        raise ValueError("Argument value error: df must contain at least one row of data")
+    elif not (('date_report' in df.columns) and ('province' in df.columns)):
+        raise ValueError("Argument value error: df must contain date_report and province columns")
+
+    df['date_report'] = pd.to_datetime(df['date_report'], format='%d-%m-%Y')
+    max_date = df.loc[df['date_report'].argmax(), 'date_report']
+    columns = ['date_report', 'province'] + list(set(df.columns) - set(['date_report', 'province']))
+    summary = df[df['date_report'] == max_date][columns].sort_values('province')
```

Comment:
I think more useful stats metrics could be added to the summary output, since returning the df filtered to the max_date does not really provide more info than was in the given df.  For example, if a specific date was given into the get_covid_data() function already, then this summary function would return the exact same df as inputted. 
I suggest maybe adding columns to the summary for information such as mean cases per day per province, max and min cases per day, or deviation of cases  etc. You can add the stat metric as a input argument to your function if you want, or just have the function return all the stat metrics.